### PR TITLE
Support private HTTP template URLs

### PR DESCRIPTION
**Pull request by jpb**
Use signed URL for `http`/`https` `Template`s to support non-public readable templates hosted on S3.

### CODE REVIEW

File: src/cfn/index.ts [hunk 1]
```diff
@@ -546,6 +582,11 @@ async function loadCFNTemplate(location: string, baseLocation: string):
   if (_.isUndefined(location)) {
     return {};
   }
+  if (location.match(/^http/) !== null) {
```
[the remark this answers is not in the document]
Comment (jpb) [reply src/cfn/index.ts]: From what I can tell, S3 doesn't support signature v2 which has a maximum one year expiry. Signature v4 has a maximum 7 day expiry. Given that signed URLs have such a short lifetime, I don't think this is an edge case we need to support right now. I think a better solution would be to give the current IAM principal access to the other account's bucket or just use a bucket in the same account.

**Comment by tavisrudd** [comment]
I'm wondering what we should do if a user explicitly passes in a pre-signed
http url and we encounter it here.

File: src/cfn/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tavisrudd) [reply src/cfn/index.ts]: Regardless of the expiration issue, I think we need to respect that other tooling might be passing one in. A check for `Signature=` in the `location` would suffice to handle that case.

[the remark this answers is not in the document]
Comment (tavisrudd) [reply src/cfn/index.ts]: I think we should also add a comment explaining why we're checking / rewriting the URL here and how this use case is different from the `s3://...` url rewrite that this function does on lines 601-608.

**Comment by garbles** [comment]
Seems good

**Comment by jpb** [comment]
It looks like this introduced a bug for rendered templates - `Template: render:stack.yaml` produces the error:

```
error: Your cloudformation Template from stack.yaml appears to use iidy's yaml pre-processor syntax.
 You need to prefix the template location with "render:".
 e.g.   Template: "render:stack.yaml"
```

**Pull request merged**